src/components/WalletPage/Swap/Swap.tsx

```diff
@@ -0,0 +1,176 @@
+  setCurrent: (action: ActionType | null) => void;
+}
+
+export default function TokenSwap({ setCurrent }: TokenSwapProps) {
+  const [payToken, setPayToken] = useState(tokens[0]);
+  const [receiveToken, setReceiveToken] = useState(tokens[1]);
+  const [payAmount, setPayAmount] = useState("");
+  const [receiveAmount, setReceiveAmount] = useState("");
+  const [fetchingQuote, setFetchingQuote] = useState(false)
+
+    ;
+
+  useEffect(() => {
+    if (!payAmount || !/^\d*\.?\d*$/.test(payAmount)) {
+      setReceiveAmount("");
+      return;
+    }
+    setFetchingQuote(true);
+
+    const fetchQuote = async () => {
+      try {
+        const res = await axios.get(`https://quote-api.jup.ag/v6/quote?inputMint=${payToken.mint}&outputMint=${receiveToken.mint}&amount=${Number(payAmount) * (10 ** payToken.decimals)}&slippageBps=50`)
+
+        const outputAmount = Number(res.data.outAmount) / 10 ** receiveToken.decimals;
+        setReceiveAmount(outputAmount.toString());
+      } catch (error) {
+        console.error("Error fetching swap price:", error);
+        setReceiveAmount("");
+      } finally {
+        setFetchingQuote(false);
+      }
+    };
+
+    const debounceFetch = setTimeout(fetchQuote, 500); // Debounce for 500ms
+    return () => clearTimeout(debounceFetch); // Clean up previous timeout if the amount changes
+  }, [payAmount, payToken, receiveToken]);
+
+  const handleSwap = () => {
+    setPayToken(receiveToken);
+    setReceiveToken(payToken);
+    setPayAmount(receiveAmount);
+    setReceiveAmount(payAmount);
+  };
+
+  useEffect(() => {
+    console.log('i have this: ', payToken, 'i want this: ', receiveToken)
+  }, [payToken, receiveToken])
+
+
+  const selectTokenValue = (value: TokensType) => {
+      const newToken = tokens.find((t) => t.symbol === value);
+      if (newToken && newToken !== receiveToken) setPayToken(newToken);
+    }
+  
+
+  return (
+    <div className="w-full max-w-md mx-auto space-y-6 bg-white rounded-lg">
+       <Top setCurrent={setCurrent}/>
+      <div className="space-y-4">
+        <div>
+          <label className="block text-sm font-medium text-gray-700 mb-1">You Pay:</label>
+          <div className="flex items-center space-x-2">
+            <Select
+              value={payToken.symbol}
+              onValueChange={(value: TokensType) => {
+                const newToken = tokens.find((t) => t.symbol === value);
+                if (newToken && newToken !== receiveToken) setPayToken(newToken);
+              }}
+            >
+              <SelectTrigger className="w-[250px]">
+                <SelectValue />
+              </SelectTrigger>
+              <SelectContent className="bg-white">
+                {tokens.map((token) => (
+                  <SelectItem key={token.symbol} value={token.symbol} disabled={token === receiveToken}>
+                    <div className="flex gap-3">
+                      <Image height={20} width={20} alt={token.name} src={token.icon || ""} />
```

Comment:
Add some default token image instead of "" src for better UX.

Reply:
I will remove the comments

Reply:
Thanks for pointing out sir @Rishabh-Singh-Codes

Reply:
done